src/main/io/osd.c

```diff
@@ -518,17 +518,20 @@ static bool osdDrawSingleElement(uint8_t item)
         }
 
     case OSD_CRAFT_NAME:
+        // This does not strictly support iterative updating if the craft name changes at run time. But since the craft name is not supposed to be changing this should not matter, and blanking the entire length of the craft name string on update will make it impossible to configure elements to be displayed on the right hand side of the craft name.
+        //TODO: When iterative updating is implemented, change this so the craft name is only printed once whenever the OSD 'flight' screen is entered.
+
         if (strlen(pilotConfig()->name) == 0) {
             strcpy(buff, "CRAFT_NAME");
         } else {
             for (unsigned int i = 0; i < MAX_NAME_LENGTH; i++) {
                 buff[i] = toupper((unsigned char)pilotConfig()->name[i]);
                 if (pilotConfig()->name[i] == 0) {
-                    buff[i] = SYM_BLANK;
+                    break;
```

Comment:
I can see two problems here.
1. The new code leaves a "0" byte in the buffer. I have not checked, that **might**  be shown as a blank space. Or something else. Safer to still translate to a `SYM_BLANK` char. 
2. Not filling `SYM_BLANK` to the end of the field might cause garbage being shown after the name. For instance when changing from a long to a short name. Would require a re-init (Power cycle?).

Reply:
1. The `\0` is intended as the string terminator. Leaving the string unterminated would be wrong, unless we know that `buff` is initialised to be filled with `\0` anyway, and even then it would be sloppy. Adding an unneeded blank at the end would definitely be wrong.

2. See comment in this pull request.